app/api/helpers/db.py

```diff
@@ -24,3 +24,15 @@ def save_to_db(item, msg="Saved to db", print_error=True):
         logging.error('DB Exception! %s' % e)
         db.session.rollback()
         return False
+
+def get_or_create(model, **kwargs):
+    was_created = False
+    instance = db.session.query(model).filter_by(**kwargs).first()
+    if instance:
+        return instance, was_created
+    else:
+        instance = model(**kwargs)
+        db.session.add(instance)
+        db.session.commit()
+        was_created = True
+        return instance, was_created
```

Comment:
Although it was there before in helpers, I believe that first() solves our purpose here. Basically, we are looking if there is one or more object is there with these properties therefore first() ensures we get at least one.